Added installation instructions and usage for the syntax visualizer to readme

MaStr11 opened the pull request:
The installation of the syntax visualizer is a constant source of confusion. As this repo is it's new home, the installation instruction should live here. I copied the documentation from the [roslyn wiki](https://github.com/dotnet/roslyn/wiki/Syntax%20Visualizer) and modified it to reflect the new installation procedure. I can't change the wiki page in roslyn directly but that page should be updated to point to this repo once this PR is merged (or should be updated if you reject this PR). I haven't copied the images yet as I'm not sure if a single readme.md is the best way to do the documentation.

Tagging @sharwell @jmarolf

sharwell commented on README.md:
```diff
@@ -4,4 +4,177 @@
+
+*Figure 1 Launching the Syntax Visualizer*
+
+![fig1.png](images/fig1.png)
```
❗️ All images are missing from the PR

MaStr11 replied:
I know. I wasn't sure if you are willing to accept the PR and even if you do it's still debatable if the documentation

* Should be all in readme.md
* Should be in a docs folder and readme.md only has the installation instructions and points to the docs in the docs folder
* Should be in the wiki

If it's all in the readme.md I can also copy the images.

sharwell replied:
Do you believe you would have found it if the documentation was here? I think I still would have had to ask a question.

sharwell replied:
📝 Note that I'm not saying this is good or bad. Perhaps it's something that works for everyone else, or maybe it's just a step in the right direction that needs other changes as well.

MaStr11 replied:
There are several issues open in roslyn and here that describe the situation and give instructions how to solve it. All of them have at least one thumb up. If people are able to find it in the issues it's clear to me many more will find it in the readme.

MaStr11 replied:
The major issue is that the documentation in the roslyn wiki and on marketplace are out of date and it is mentioned nowhere how to get the syntax visualizer (even you needed to ask). I just wanted to help out with updating the docs and as I'm not able to contribute to the roslyn wiki and this is the better place anyway I copied the documentation and actualized the installation instructions. I think there is nothing wrong with accepting the PR and dropping it in favor of something else (docs.microsoft.com?) later.

MaStr11 replied:
> or maybe it's just a step in the right direction that needs other changes as well

There are only two places where the syntax visualizer is mentioned. These two places need to be updated:

* [roslyn wiki](https://github.com/dotnet/roslyn/wiki/Syntax%20Visualizer)
* [Marketplace](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.NETCompilerPlatformSDK)

BillWagner commented:
A few thoughts here:

I like what @MaStr11 has written. We don't have this kind of detail on the docs site yet regarding the syntax visualizer and what it can do.  I think it would help a lot of customers if it were there.

I've got a start on docs related to building analyzers [under the C# guide](https://docs.microsoft.com/dotnet/csharp/roslyn-sdk/). That seems like a good place for this information. (Note there are two open PRs that add to what's there currently: /dotnet/docs#4289 and /dotnet/docs#4331) 

A proposal: Add a new "How to install the SDK and explore code models" as a first article under the roslyn-sdk page I mentioned above. 

An alternative location in docs would be under the Visual Studio content. @gewarren would be the person to comment on where it might live in that set. From either location, we can cross-link as needed.

MaStr11 commented:
I wasn't aware that there is already roslyn related content on docs.microsoft.com. That's great and should be mentioned in the roslyn wiki and readme. The documentation for the visualizer should definitely live there. I would propose to alter this PR to include only the installation instructions and a link to docs.microsoft.com. The SDK help page at docs.microsoft.com should also start with the installation instructions. The documentation for the visualizer is quite long so it might be better to make it a page for its own (btw it wasn't created by me. I just copied it).

jinujoseph commented:
Thanks @BillWagner , I think a session "How to install the SDK and explore code models" sounds good with the contents here.  I agree that all information should point to single docs.microsoft.com page from this repro, from roslyn wiki etc

MaStr11 commented:
I removed the documentation of the syntax visualizer from the readme (it can still be found [here](https://github.com/dotnet/roslyn/wiki/Syntax%20Visualizer)). I also added a brief summary about Roslyn which gave me the possibility to link to the Roslyn repo, the documentation at docs.microsoft.com and to explain how this repo is different from Roslyn. @BillWagner will you move the Syntax Visualizer documentation to docs.microsoft.com?

BillWagner commented:
Based on discussion, I created dotnet/roslyn-api-docs#36 to create this article in the C# guide docs.

@sharwell @jinujoseph @MaStr11 should we close this now?

@MaStr11 thanks for bringing this to our attention.

MaStr11 commented:
I already removed the documentation of the visualizer in 3a09de6. What's left are the installation instructions and some context information with links to docs.microsoft.com. 

> should we close this now?

If *closing* means *merging* then yes.

BillWagner commented:
Sorry @MaStr11 Yes, *merge*, not *close*.

jinujoseph commented:
Thanks @BillWagner and @MaStr11
